srcpkgs/ldb/template

```diff
@@ -1,29 +1,76 @@
 # Template file for 'ldb'
 pkgname=ldb
-version=1.5.1
+version=2.2.0
 revision=1
-build_style=gnu-configure
-configure_args="--with-modulesdir=/usr/lib/ldb/modules"
-hostmakedepends="python pkg-config docbook-xsl libxslt"
-makedepends="tdb-devel tevent-devel popt-devel talloc-devel lmdb-devel"
+build_style=waf3
+build_helper="qemu"
+configure_script="buildtools/bin/waf"
+configure_args="--sysconfdir=/etc --localstatedir=/var
+ --disable-rpath --disable-rpath-install --without-gettext
+ --builtin-libraries=replace --bundled-libraries=NONE
+ --with-modulesdir=/usr/lib/ldb/modules"
+hostmakedepends="pkg-config docbook2x tdb-python3 tevent-python3"
+makedepends="python3-devel tdb-devel tevent-devel talloc-devel
+ popt-devel lmdb-devel libxslt gettext-devel cmocka-devel"
 checkdepends="cmocka-devel"
 short_desc="LDAP-like database from samba"
 maintainer="Yuusha Spacewolf <xyuusha@paranoici.org>"
 license="LGPL-3.0-or-later"
 homepage="https://www.samba.org/ldb/"
 distfiles="https://www.samba.org/ftp/pub/ldb/ldb-${version}.tar.gz"
-checksum=232e54b87c53210a6861424ad411b04dbfa0f2ca4bc463aefea188158e0cc963
+checksum=134bb51769709af59f30bf468e454d1377a8096acd4e80dcb42fd264f558bd5f
 
-nocross="see srcpkgs/tdb (cross-32bit.answers)."
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --cross-compile --hostcc=cc
+	 --cross-execute=qemu-${XBPS_TARGET_QEMU_MACHINE}-static"
+fi
 
 # workaround for cmocka's broken uintptr_t definition on musl
 if [ "$XBPS_TARGET_WORDSIZE" = "64" -a "$XBPS_TARGET_LIBC" = "musl" ]; then
 	CFLAGS+=" -D__WORDSIZE=64"
 fi
 
+post_extract() {
+	# wrap target python3.9-config to find the right python flags
+	if [ "$CROSS_BUILD" ]; then
+		local _pyconfig="${XBPS_WRAPPERDIR}/python${py3_ver}-config"
+		cat> "${_pyconfig}" <<-EOF
+		#!/bin/sh
+		exec "${XBPS_CROSS_BASE}/usr/bin/python${py3_ver}-config" "\$@"
+		EOF
+		chmod 755 "${_pyconfig}"
+	fi
+}
+
+post_patch() {
+	# Avoid Python shlib extension noise in libpyldb-util.so
+	vsed -e "/env.pyext_PATTERN/s/dct\['SO'\]/'.so'/" \
+		-i third_party/waf/waflib/Tools/python.py
+
+	if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
+		# ldb_match_test.c fails to compile on musl; disable it
+		patch -Np0 < ${FILESDIR}/disable-compile-error-test.patch
+	fi
```

Comment:
If the test works for glibc, I'd prefer to disable it for all targets... Alternatively, fix the test on musl (probably a bad idea). Simplifies the template.

Reply:
I went with sgn's suggestion.

Reply:
Yup, I prefer his.